protocol_rfcs/managed-commits.md

```diff
@@ -0,0 +1,315 @@
+    - The commit-owner must track these files until they are backfilled.
+    - Mere existence of these files does not mean that the file is a _valid_ commit. It might correspond to a failed or in-progress commit.
+      The commit-owner is the source of truth around which un-backfilled commits are valid.
+
+2. **Backfilled commit files**: These reside in the `_delta_log` directory.
+    - The filename must follow the pattern: `<version>.json`. Here the `version` is the version `v` which is being committed, zero-padded to 20 digits.
+    - The existence of a `<version>.json` file proves that the corresponding version `v` is committed, even for managed-commit tables. Filesystem based Delta clients can use filesystem listing operations to directly discover such commits.
+
+Without [managed-commits](#managed-commits), a delta client must always write backfilled commit files directly, relying on filesystem atomicity
+to prevent lost writes when multiple writers attempt to commit the same version at the same time.
+
+With [managed-commits](#managed-commits), the delta client asks the [commit-owner](#commit-owner) to commit the version `v` and the commit-owner
+decides whether to write the backfilled/un-backfilled commit file based on the [managed commit protocol](#commit-protocol).
+
+## Commit Owner API
+
+When managed commits are enabled, a `commit-owner` manages commits to the table on behalf of the Delta client. A commit-owner always has a client-side component (which the Delta client interacts with directly). It may also
+involve a server-side component (which the client-side component would be responsible to communicate with). The Delta client is responsible to define the client-side API that commit-owners should target, and commit-owners
+are responsible to define the commit atomicity and backfill protocols which the commit-owner client should implement.
+
+At a high level, the `commit-owner` needs to provide:
+- API to atomically commit a version `x` with given set of `actions`. This is explained in detail in the [commit protocol](#commit-protocol) section.
+- API to retrieve information about the recent commits on the table. This is explained in detail in the [getting un-backfilled commits from commit-owner](#getting-un-backfilled-commits-from-commit-owner) section.
+
+### Commit Protocol
+
+When a `commit-owner` receives a request to commit version `v`, it must first verify that the previous version `v-1` already exists, and that version `v` does not yet exist. It then has following choices to publish the commit:
+1. Write the actions to an 'un-backfilled' [commit file](#commit-files) in the `_delta_log/_commits` directory, and **atomically** record that the new file now corresponds to version `v`.
+2. Atomically write a backfilled [commit file](#commit-files) in the `_delta_log` directory. Note that the commit will be considered to have succeeded as soon as the file becomes visible to
```

Comment:
`Write the delta file to the _delta_log/_commits directory` , `Atomically write the delta file to the _delta_log directory` implicitly assumes that the file inside _commits directory must have UUID while files inside `_delta_log` won't have UUID. This is true but it might be better to explicitly mention this by using backfill/un-backfilled commit words?